pkg/mimir/modules.go

```diff
@@ -883,10 +883,14 @@ func (t *Mimir) initRuler() (serv services.Service, err error) {
 }
 
 func (t *Mimir) initAlertManager() (serv services.Service, err error) {
-	// Initialize the compat package in classic mode. This has the same behavior as if
-	// the compat package was not initialized, but with debug logs when a configuration
-	// is loaded or a new configuration is submitted.
-	features, err := featurecontrol.NewFlags(util_log.Logger, featurecontrol.FeatureClassicMode)
+	mode := featurecontrol.FeatureClassicMode
+	if t.Cfg.Alertmanager.UTF8StrictMode {
+		level.Debug(util_log.Logger).Log("msg", "Starting Alertmanager in UTF-8 strict mode")
+		mode = featurecontrol.FeatureUTF8StrictMode
+	} else {
+		level.Debug(util_log.Logger).Log("msg", "Starting Alertmanager in classic mode")
+	}
+	features, err := featurecontrol.NewFlags(util_log.Logger, mode)
```

Comment:
Nit, this is pretty important and only logged once - I'd consider changing his to info. 

```suggestion
	mode := featurecontrol.FeatureClassicMode
	if t.Cfg.Alertmanager.UTF8StrictMode {
		level.Info(util_log.Logger).Log("msg", "Starting Alertmanager in UTF-8 strict mode")
		mode = featurecontrol.FeatureUTF8StrictMode
	} else {
		level.Info(util_log.Logger).Log("msg", "Starting Alertmanager in classic mode")
	}
	features, err := featurecontrol.NewFlags(util_log.Logger, mode)
```